utils/build-script-impl

```diff
@@ -1491,19 +1516,26 @@ function set_swiftpm_bootstrap_command() {
 
 
 for host in "${ALL_HOSTS[@]}"; do
+    # Skip this pass when the only action to execute can't match.
+    if [[ "${ONLY_EXECUTE}" != "" ]] && ! [[ "${ONLY_EXECUTE}" == ${host}-*-build ]]; then
+        continue
+    fi
 
     calculate_targets_for_host $host
 
     set_build_options_for_host $host
 
-    echo "Building the standard library for: ${SWIFT_STDLIB_TARGETS[@]}"
-    if [[ "${SWIFT_TEST_TARGETS[@]}" ]] && ! [[ "${SKIP_TEST_SWIFT}" ]]; then
-        echo "Running Swift tests for: ${SWIFT_TEST_TARGETS[@]}"
-    fi
-    if ! [[ "${SKIP_TEST_BENCHMARKS}" ]] &&
-       [[ "${SWIFT_RUN_BENCHMARK_TARGETS[@]}" ]] &&
-       ! [[ "${SKIP_TEST_BENCHMARK}" ]]; then
-        echo "Running Swift benchmarks for: ${SWIFT_RUN_BENCHMARK_TARGETS[@]}"
+    # Don't echo anything if only executing an individual action.
```

Comment:
I'm afraid the build-script does not have this information yet, it does not know the specific targets that will be executed.  (It is important to know whether we will be doing short vs. validation vs. long tests for example, or which platforms the tests will be run for.)  Did you plan to move this computation over as well?